core/server/api/canary/utils/serializers/output/utils/members.js

```diff
@@ -1,45 +1,42 @@
+const _ = require('lodash');
 const labs = require('../../../../../../services/labs');
-const membersService = require('../../../../../../services/members');
-const MEMBER_TAG = '#members';
-const PERMIT_CONTENT = false;
-const BLOCK_CONTENT = true;
-
-// Checks if request should hide memnbers only content
-function hideMembersOnlyContent(attrs, frame) {
-    const membersEnabled = labs.isSet('members');
-    if (!membersEnabled) {
+
+// Checks if request should hide members only content
+function hideMembersOnlyContent({visibility}, frame) {
+    const PERMIT_CONTENT = false;
+    const BLOCK_CONTENT = true;
+
+    if (visibility === 'public') {
         return PERMIT_CONTENT;
     }
 
-    const postHasMemberTag = attrs.tags && attrs.tags.find((tag) => {
-        return (tag.name === MEMBER_TAG);
-    });
     const requestFromMember = frame.original.context.member;
-    if (!postHasMemberTag) {
-        return PERMIT_CONTENT;
-    }
+
     if (!requestFromMember) {
         return BLOCK_CONTENT;
-    }
-
-    const memberHasPlan = !!(frame.original.context.member.stripe.subscriptions || []).length;
-    if (!membersService.isPaymentConfigured()) {
+    } else if (visibility === 'members') {
         return PERMIT_CONTENT;
     }
-    if (memberHasPlan) {
+
+    const memberHasPlan = !!(_.get(frame, 'original.context.member.stripe.subscriptions', [])).length;
+
+    if (visibility === 'paid' && memberHasPlan) {
         return PERMIT_CONTENT;
     }
+
     return BLOCK_CONTENT;
 }
 
 const forPost = (attrs, frame) => {
-    const hideFormatsData = hideMembersOnlyContent(attrs, frame);
-    if (hideFormatsData) {
-        ['plaintext', 'html'].forEach((field) => {
-            attrs[field] = '';
-        });
-    }
     if (labs.isSet('members')) {
+        const hideFormatsData = hideMembersOnlyContent(attrs, frame);
```

Comment:
Moving into this block means that the visibility attribute will only take affect when members is on - that is most likely correct and you just fixed a bug :sweat_smile: Maybe one to check w/ zimo/john tho?